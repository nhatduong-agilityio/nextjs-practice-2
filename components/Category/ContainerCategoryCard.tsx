// Libs
import type { SystemStyleObject } from '@chakra-ui/react'
import { Box, Flex } from '@chakra-ui/react'
import type { ReactNode } from 'react'
import React from 'react'
import type { StaticImageData } from 'next/legacy/image'
import Image from 'next/legacy/image'

interface IProps {
  sxContainer?: SystemStyleObject
  sxWrapperImage?: SystemStyleObject
  srcImage: StaticImageData
  imageAlt: string
  sxWrapperChildren?: SystemStyleObject
  children?: ReactNode
}

const ContainerCardCategory = ({
  sxContainer,
  sxWrapperImage,
  srcImage,
  imageAlt,
  sxWrapperChildren,
  children,
}: IProps) => (
  <Box sx={sxContainer}>
    <Box position='relative' zIndex='1' w='full' sx={sxWrapperImage}>
      <Image src={srcImage} alt={imageAlt} sizes='100vw' layout='fill' objectFit='cover' priority />
      <Flex position='absolute' zIndex='2' w='full' height='100%' sx={sxWrapperChildren}>
        {children}
      </Flex>
    </Box>
  </Box>
)

export default ContainerCardCategory
