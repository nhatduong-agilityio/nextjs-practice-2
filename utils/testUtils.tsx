import { ChakraProvider } from '@chakra-ui/react'
import { AuthProvider } from '@store/providers/AuthProvider'
import { LoadingProvider } from '@store/providers/LoadingProvider'
import type { RenderOptions, RenderResult } from '@testing-library/react'
import { render } from '@testing-library/react'
import { customTheme } from '@themes/index'
import type { ReactElement } from 'react'

const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
  return (
    <ChakraProvider theme={customTheme}>
      <LoadingProvider>
        <AuthProvider>{children}</AuthProvider>
      </LoadingProvider>
    </ChakraProvider>
  )
}

const customRender = (ui: ReactElement, options?: Omit<RenderOptions, 'wrapper'>): RenderResult =>
  render(ui, { wrapper: AllTheProviders, ...options })

export * from '@testing-library/react'
export { customRender as render }
