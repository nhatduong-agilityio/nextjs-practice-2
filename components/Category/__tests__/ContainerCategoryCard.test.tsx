// Utils
import { render } from '@utils/testUtils'

import { CategoriesMock } from '@mock/dataMock'

// Components
import ContainerCategoryCard from '../ContainerCategoryCard'

describe('ContainerCategoryCard render', () => {
  const propsMock = {
    srcImage: CategoriesMock.CHAIR.OFFICE.BG_IMAGE,
    imageAlt: CategoriesMock.CHAIR.OFFICE.IMAGE_ALT,
    name: CategoriesMock.CHAIR.OFFICE.NAME,
    pathLink: '/',
  }

  it('Should ContainerCategoryCard match Snapshot', () => {
    const { container } = render(
      <ContainerCategoryCard srcImage={propsMock.srcImage} imageAlt={propsMock.imageAlt} />,
    )

    expect(container).toMatchSnapshot()
  })
})
