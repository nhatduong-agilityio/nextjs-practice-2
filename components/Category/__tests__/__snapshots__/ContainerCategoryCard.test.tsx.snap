// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContainerCategoryCard render Should ContainerCategoryCard match Snapshot 1`] = `
<div>
  <div
    class="css-0"
  >
    <div
      class="css-1y79m5s"
    >
      <span
        style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
      >
        <img
          alt="Picture of furniture office chair category"
          data-nimg="fill"
          decoding="async"
          sizes="100vw"
          src="/_next/image?url=%2Fimg.jpg&w=3840&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=640&q=75 640w, /_next/image?url=%2Fimg.jpg&w=750&q=75 750w, /_next/image?url=%2Fimg.jpg&w=828&q=75 828w, /_next/image?url=%2Fimg.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimg.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimg.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimg.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimg.jpg&w=3840&q=75 3840w"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover;"
        />
      </span>
      <div
        class="css-1ko0w9n"
      />
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
